runtime/testenv/testenv.go

```diff
@@ -118,7 +111,32 @@ func WithCRDPath(path ...string) Option {
 //
 // NOTE: This function should be called only once for each package you are running tests within, usually the environment
 // is initialised in a suite_test.go or <package>_test.go file within a `TestMain` function.
+//
+// When a testenv Environment is created, it initializes the
+// controller-runtime's deferred logger with a default logger based on klog. In
+// order to override this behavior, the controller-runtime logger can be
+// initialized before creating testenv Environment.
+//
+//  import (
+//      "testing"
+//
+//      "github.com/fluxcd/pkg/runtime/testenv"
+//      ctrl "sigs.k8s.io/controller-runtime"
+//      "sigs.k8s.io/controller-runtime/pkg/log/zap"
+//  }
+//
+//  func TestMain(m *testing.M) {
+//      zlog := zap.New(zap.UseDevMode(true))
+//      ctrl.SetLogger(zlog)
+//
+//      testEnv = testenv.New()
+//      ...
+//  }
+//
 func New(o ...Option) *Environment {
+	// Set a default logger if not set already.
+	log.SetLogger(klogr.New())
```

Comment:
If this is equivalent to ctrl.SetLogger, it'd be better to use the same in both the example given and the code. (And if it's not, why do they differ?)